core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -2615,7 +2615,10 @@ class ReplicaManagerTest {
 
   @Test
   def testStopReplicaWithDeletePartitionAndExistingPartitionAndNewerLeaderEpochAndIOException(): Unit = {
-    testStopReplicaWithExistingPartition(2, true, true, Errors.KAFKA_STORAGE_ERROR)
+    // Even though we are trying to trigger an IOException by deleting the underlying log directory,
+    // given the async deletion of a replica no longer needs the underlying directory, the stopReplica operation
```

Comment:
This is not very precise. The async deletion still needs the underlying dir. It's just that it happens asynchronously and won't trigger an exception during the synchronous call.

Reply:
Please check my reply below and see if that makes sense.